src/main/java/server/request/RequestParser.java

```diff
@@ -0,0 +1,37 @@
+package main.java.server.request;
+
+import java.util.HashMap;
+
+public class RequestParser {
+
+    private String[] segmentedRequest;
+    private final HashMap<String, String> headers = new HashMap<>();
```

Comment:
I think getting rid of these variables might clean things up a bit.
One weird thing about them, which might not matter in this case, is that if you reuse an instance of `RequestParser` later, those variables probably still have the old data.